File: smkn8picket-client/src/models/Requests/StudentProgressNoteRequest.ts
import type { StudentProgressNoteType } from "../Enums/StudentProgressNoteType"

export default interface StudentProgressNoteRequest {
  id: number
  studentId: number
  teacherId: number
  note: string
  createdAt: Date | string
  updatedAt?: Date
  progressType: StudentProgressNoteType
}

File: SharedModel/Requests/StudentProgressNoteRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SharedModel.Requests
{
    public record StudentProgressNoteRequest(
        int StudentId,
        int TeacherId,
        int SchoolYearId,
        string Note,
          StudentProgressNoteType ProgressType
        );

}
